packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -537,6 +590,7 @@ class ReactSixteenAdapter extends EnzymeAdapter {
         const name = type.render.displayName || functionName(type.render);
         return name ? `ForwardRef(${name})` : 'ForwardRef';
       }
+      case Lazy || NaN: return 'LazyComponent';
```

Comment:
`React.lazy` doesn't seem to provide a display name - let's use:
```suggestion
      case Lazy || NaN: {
        if (type.displayName) {
          return type.displayName;
        }
        const name = displayNameOfNode({ type: type._ctor });
        return name ? `React.lazy(${name})` : `lazy`;
      }
```
let's also add tests for an explicit `Object.assign(React.lazy(Component), { displayName: 'something' })`, as well as a lazy component wrapping something *with* a displayName, one without, and one with no name at all.

Reply:
It's tricky to test if displayName of wrapped component is included in displayName of lazy component. The `_ctor` is not the wrapped component itself but the function dynamically importing wrapped component. Before it's loaded we can't get the displayName of wrapped component.

Reply:
interesting, ok - that actually does make sense, considering that one of the use cases for `lazy` is a component that hasn't been fetched yet.